sdk/java-sdk/src/main/java/kalix/javasdk/DeferredCall.java

```diff
@@ -41,4 +42,10 @@
    * effects().asyncReply} and {@code effects().asyncEffect}
    */
   CompletionStage<O> execute();
+
+  /**
+   * Add given entries to metadata
+   * @return DeferredCall with updated metadata
+   */
+  DeferredCall<I, O> withMetadata(List<Metadata.MetadataEntry> entries);
```

Comment:
Not sure about the naming: `withMetadata` vs `addMetadata`. Current semantics is about adding.

Reply:
I think it should be `withMetadata(Metadata)` rather than a list of entries. I think we provide APIs for manipulating that metadata elsewhere. (In general across the Akka and Kalix APIs we use `withX` to replace and `addX` for update/add)

I expect the main use scenario will be "do this deferred call with this set of metadata", rather than iteratively adding metadata entries to one deferred call, so optimize DX for that:

Reply:
That was my first idea, but the implementation is hidden and I cannot use it `private[kalix] class MetadataImpl`. Implementing `Metadata.MetadataEntry` is faster than `Metadata` interface, but maybe we should simply expose some classes, as suggested in one of my comments.

Reply:
I don't understand this comment. `kalix.scalasdk.Metadata` and `kalix.javasdk.Metadata` are public APIs for dealing with metadata. A user would do `deferredCall.withMetadata(Metadata.empty.add("some-key", "some value"))` in Scala and `deferredCall.withMetadata(Metadata.EMPTY.add("some-key", "some value"))` in Java.

The internal impl would be done by dealing with the underlying MetadataImpl turning it into gRPC/protocol metadata just like for responses in for example `ActionsImpl#toProtocol`?

Reply:
🤦 of course, fixed